lib/spack/spack/spec.py

```diff
@@ -4433,10 +4423,8 @@ def splice(self, other, transitive):
                 # are cached writing specs only writes cached hashes in case
                 # the spec is too old to have full provenance for these hashes,
                 # so we can't rely on doing it at write time.
-                if has_build_hash:
-                    _ = dep.build_hash()
-                if has_full_hash:
-                    _ = dep.full_hash()
+                if has_dag_hash:
+                    _ = dep.dag_hash()
```

Comment:
Why do we assign this value? We should just call `dag_hash()`:
```suggestion
                if has_dag_hash:
                    dep.dag_hash()
```